File: stack.py
class Stack :
    def __init__(self, top=0):
        self.top = top
        self.arr = [] 
    def __str__(self):
        return str(self.arr)

    def push(self, val):
        self.arr.append(val)
        self.top+=1

    def pop(self):
        if len(self.arr) == 0:
            return None
        v = self.arr[len(self.arr)-1]
        self.arr = self.arr[:len(self.arr)-1]
        return v 


s = Stack()
s.push(1)
s.push(2)
s.push(3)
print(s)
s.pop()
print(s)
s.pop()
print(s)

File: stack.js

function Stack() {
    this.arr = [];
    this.top = -1;
    this.init = init;
    this.push = push;
    this.pop = pop;
    this.size = size;
    this.toString = toString;
}

function init() {
    this.arr = [];
    this.top = -1;
}

function push(e) {
    this.arr[++this.top] = e;
}

function pop() {
    if (this.size() < 1) {
        return NaN
    }
    r = this.arr[this.top]
    this.arr = this.arr.slice(0, this.arr.length-1)
    this.top--
    return r
}

function size() {
    return this.arr.length;
}

function toString() {
    return this.arr.toString();
}

s = new Stack();
s.push(1);
s.push(2);
s.push(3);
s.push(4);
console.log(s.toString());
console.log(s.pop());
console.log(s.toString());
console.log(s.pop());
console.log(s.toString());

s.init();
console.log(s.toString());
